x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -366,6 +341,13 @@ public void putJob(PutJobAction.Request request, AnalysisRegistry analysisRegist
             return;
         }
 
+        // Check the job id is not the same as a group Id
+        if (currentMlMetadata.isGroupOrJob(job.getId())) {
```

Comment:
Are we also validating this when the job is written to the index?

Reply:
There are 2 checks [here](https://github.com/elastic/elasticsearch/blob/d6f7b9bf22ab78f61db1b0a903359d13afd7441f/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java#L410) 

1. that the job id is not a group
2. that any groups the job has are not job Ids

I realised the second was missing so I pushed another commit with that. Previously those checks were done in the constructor of `MlMetadata` which further made me realise I need to perform the same check on job update so I made that change also.